src/megatron/energon/av/decode_av_frames.py

```diff
@@ -0,0 +1,441 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import io
+from collections.abc import Collection, Iterator
+from fractions import Fraction
+
+# Try importing optional dependencies
+try:
+    import av
+    import numpy as np
+    import soundfile as sf
+    import torch
+    import filetype
+    from megatron.energon.av.fastseek import Fastseek
+    from megatron.energon.av.video_data import VideoData
+    AV_DECODE_AVAILABLE = True
+except ImportError as e:
+    AV_DECODE_AVAILABLE = False
+    MISSING_DEPENDENCY = str(e)
+
+DEFAULT_AUDIO_FRAME_SHIFT_MS = 10  # in milliseconds
+
+class AVData:
```

Comment:
I'd suggest to make this class the only entrypoint user-facing. (And the `AVDecoder` for use with webdataset)

Reply:
done